desktopexporter/internal/server/server_test.go

```diff
@@ -0,0 +1,321 @@
+			Version:                "1",
+			Attributes:             map[string]any{},
+			DroppedAttributesCount: 0,
+		},
+		DroppedAttributesCount: 0,
+		DroppedEventsCount:     0,
+		DroppedLinksCount:      0,
+		StatusCode:             "",
+		StatusMessage:          "",
+	}
+
+	if err := s.Store.AddSpans(context.Background(), []telemetry.SpanData{testSpanData}); err != nil {
+		t.Fatalf("could not create test span: %v", err)
+	}
+	return s, func(t *testing.T) {
+		s.Store.Close()
+	}
+}
+func TestTracesHandler(t *testing.T) {
+	t.Run("Traces Handler (Empty)", func(t *testing.T) {
+		s := NewServer("localhost:8000")
+		defer s.Store.Close()
+
+		testSummaries := telemetry.TraceSummaries{
+			TraceSummaries: []telemetry.TraceSummary{
+				{
+					HasRootSpan:     true,
+					RootServiceName: "groot",
+					RootName:        "i.am.groot",
+					RootStartTime:   time.Now(),
+					RootEndTime:     time.Now().Add(time.Minute),
+					SpanCount:       2,
+					TraceID:         "12345",
+				},
+			},
+		}
+
+		req, err := http.NewRequest("GET", "/api/traces", nil)
+		if err != nil {
+			t.Fatalf("could not create request: %v", err)
+		}
+
+		rec := httptest.NewRecorder()
+		s.tracesHandler(rec, req)
+		res := rec.Result()
+		defer res.Body.Close()
+
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			t.Fatalf("could not read response: %v", err)
+		}
+
+		if res.StatusCode != http.StatusOK {
+			t.Errorf("expected status OK; got %v", res.Status)
+		}
+
+		err = json.Unmarshal(b, &testSummaries)
+		if err != nil {
+			t.Fatalf("could not unmarshal bytes to trace summaries: %v", err.Error())
+		}
+
+		assert.Len(t, testSummaries.TraceSummaries, 0)
+	})
+
+	t.Run("Traces Handler (Not Empty)", func(t *testing.T) {
+		testSummaries := telemetry.TraceSummaries{}
+		s, teardown := setupWithTrace(t)
+		defer teardown(t)
+
+		req, err := http.NewRequest("GET", "/api/traces", nil)
+		if err != nil {
+			t.Fatalf("could not create request: %v", err)
+		}
+
+		rec := httptest.NewRecorder()
+		s.tracesHandler(rec, req)
+		res := rec.Result()
+		defer res.Body.Close()
+
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			t.Fatalf("could not read response: %v", err)
+		}
+
+		if res.StatusCode != http.StatusOK {
+			t.Errorf("expected status OK; got %v", res.Status)
+		}
+
+		err = json.Unmarshal(b, &testSummaries)
+		if err != nil {
+			t.Fatalf("could not unmarshal bytes to trace summaries: %v", err.Error())
+		}
+
+		assert.Equal(t, "1234567890", testSummaries.TraceSummaries[0].TraceID)
+		assert.Equal(t, true, testSummaries.TraceSummaries[0].HasRootSpan)
+		assert.Equal(t, "test", testSummaries.TraceSummaries[0].RootName)
+		assert.Equal(t, "pumpkin.pie", testSummaries.TraceSummaries[0].RootServiceName)
+		assert.Equal(t, uint32(1), testSummaries.TraceSummaries[0].SpanCount)
+	})
+}
+
+func TestTraceIDHandler(t *testing.T) {
+	s, teardown := setupWithTrace(t)
+	defer teardown(t)
+
+	srv := httptest.NewServer(s.Handler(false))
+	defer srv.Close()
```

Comment:
Right now this test is slightly different from the others, but they should probably all be the same. I think you could simplify things by moving these bits into `setupWithTrace(t)`. Maybe have it return both `*Server` and `*httptest.Server`

```go
server, testServer, teardown := setupWithTrace(t)
// also calls `testServer.Close()`
defer teardown(t)
```

Reply:
Good call